src/component.js

```diff
@@ -228,9 +176,7 @@ function process() {
 		rerenderQueue = [];
 		// Don't update `renderCount` yet. Keep its value non-zero to prevent unnecessary
 		// process() calls from getting scheduled while `queue` is still being consumed.
-		queue.some(c => {
-			if (c._internal._flags & DIRTY_BIT) rerenderComponent(c);
-		});
+		queue.some(rerenderComponent);
```

Comment:
I moved the dirty check into `rerenderComponent` and combined it with the unmounting check